### PR TITLE
feat: PDF export and breakdown table improvements

### LINKED ISSUE #130: [FEATURE] PDF Improvements
## Problem Statement

Currently the PDF is not that nicely styled (very boxy!). Also, some clients we might not want to give financials (even their costs) as it might go through a partner.

## Proposed Solution

- Apply a new style with background colours on boxes vs outlines
- Add drop down options to the PDF Export button of "with financials" and "without financials"

## Alternatives Considered

Can't think of another option

## Additional Context

Consider sending an e-mail of a project update

### LINKED ISSUE #132: Breakdown table: Add Budget vs Actual financial columns
## Summary

Add financial columns to the breakdown table that accurately reflect BC's billing model.

## Current State

The breakdown table shows hours worked by task/team. Previously had Unit Price and Total Price columns, but these were removed because:
- Unit Price display was confusing (task-level vs resource-level rates)
- Fixed Price projects showed irrelevant hourly rates
- Didn't distinguish between Budget vs Actual

## Proposed Columns

| Column | Source | Description |
|--------|--------|-------------|
| Hours | Timesheets | All hours (pending + approved) |
| Posted | Job Ledger Entry | Hours posted to ledger |
| Budget | Job Planning Lines | Planned/budgeted hours |
| Unit Price | Resource/Planning Lines | T&M billing rate |
| Total Price | Calculated/Ledger | Revenue |

### T&M Projects
- Show Unit Price and Total Price
- If all resources on a task have same rate → show at task level
- If different rates → show at resource level, N/A at task level

### Fixed Price Projects
- Hide Unit Price column (billing not based on hours)
- Show Budget Hours and Fixed Price from Item lines

### Data Required

1. Budget hours per task (from Job Planning Lines)
2. Posted hours/price per task (from Job Ledger Entry)
3. Rate consistency detection (same rate for all resources on task?)

## BC Setup for Block Hours

For "Fixed Price Variable Scope" (block hours):
1. Resource Billable lines - Define block (50hrs × £100 = £5,000)
2. Resource Budget lines - Allocate hours per task
3. Shows as T&M in Thyme with Unit Price visibility

## Files to Modify

- `src/services/bc/projectDetailsService.ts` - Add budget/posted per task
- `src/components/projects/ProjectTasksTable.tsx` - Conditional columns

### LINKED ISSUE #142: [ENHANCEMENT] Use Consistent naming for report
## Proposed Enhancement

Use Consistent naming for report : Customer - Project - Report

### PR DESCRIPTION
**Pull request by BenGWeeks**
## Summary

- Improve PDF export with financials options
- Remove confusing financial columns from breakdown table
- Fix unit conversion for DAY-based resources (hours saved as days)
- Plan screen UX improvements
- Consistent PDF filename: Customer - Project - Report

### PDF Export (#130)
- Added financials display options for PDF export

### Consistent Report Naming (#142)
- PDF filename now uses pattern: Customer - Project Code - Project Report
- Gracefully handles missing customer name (falls back to "Project Code - Project Report")
- Restores original page title after print dialog closes

### Breakdown Table
- Removed Unit Price and Total Price columns (were confusing)
- Keep Hours, Approved, and Pending columns

### Unit Conversion Fix
- BC uses the resource's base unit (DAY or HOUR), ignoring `unitOfMeasureCode` parameter
- Now convert hours to days before saving (e.g., 4 hours ÷ 7.5 = 0.533 days)
- Reading always converts based on resource's HOUR factor (don't trust API's unitOfMeasureCode)
- Fixes Time Budgeted and Hours per Week chart showing wrong values

### Plan Screen UX
- Add yellow dotted outline on hover for editable task rows
- Fix wrong week bug: clicking on a cell now opens the correct week's edit modal
- Chart legend now fits inside container

## Related Issues

- Closes #130 - PDF Improvements
- Fixes #142 - Consistent report naming
- Closes #132 - Documents proper financial column reimplementation
- #133 - Three-level hierarchy (Task/Role/User) for future
- thyme-bc-extension#29 - Unit of measure conversion

## Test Plan

- [ ] PDF export works with financials options
- [ ] PDF filename follows "Customer - Project Code - Project Report" pattern
- [ ] PDF filename works without customer name (just "Project Code - Project Report")
- [ ] Browser tab title restores after print dialog closes/cancels
- [ ] Breakdown table shows only Hours/Approved/Pending
- [ ] Create planning entry with 4 hours → BC shows ~0.53 quantity (for DAY-based resource)
- [ ] Time Budgeted and chart planned hours display correctly
- [ ] Yellow highlight appears on task rows when hovering
- [ ] Clicking on week cell opens correct week in edit modal

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by EdiWeeks** [comment]
@BenGWeeks do you want me to fix co-pilots comments or you prefer to do that?

File: src/services/bc/projectDetailsService.ts [hunk 1]
```diff
@@ -378,14 +384,55 @@ export const projectDetailsService = {
...
+        resourceNo: string,
+        quantity: number,
+        _unitOfMeasureCode?: string // Kept for backwards compatibility but not used
+      ): number => {
+        // Check if resource is DAY-based by looking for HOUR conversion factor > 1
+        const hourKey = `${resourceNo}:HOUR`;
+        const hourFactor = uomConversionMap.get(hourKey);
+        if (hourFactor !== undefined && hourFactor > 1) {
+          // Resource is DAY-based: quantity × hourFactor = hours
+          return quantity * hourFactor;
```
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply src/services/bc/projectDetailsService.ts]: **Update:** Fixed in c9d1f79 to support both patterns.

File: src/utils/unitConversion.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * If the resource is HOUR-based, the quantity is already in hours.
+ *
+ * NOTE: We don't trust unitOfMeasureCode from the API because BC may ignore it
+ * when creating lines.
+ */
+export function convertToHours(
+  resourceNo: string,
+  quantity: number,
+  uomConversionMap: UOMConversionMap
+): number {
```
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply src/utils/unitConversion.ts]: ✅ Fixed in commit 355cbd6 - Added 22 Vitest unit tests in `tests/unit/utils/unitConversion.test.ts` covering all 5 exported functions: `buildUOMConversionMap`, `getHoursPerDay`, `convertToHours`, `convertFromHours`, and `isResourceDayBased`. Tests cover HOUR > 1, HOUR < 1, zero/negative guards, round-trip conversion, and edge cases.
